fedimint-server/src/config/mod.rs

```diff
@@ -519,6 +578,41 @@ impl ServerConfig {
             module_cfgs.insert(module_id, cfg);
         }
 
+        let cfg = ServerConfig::from(
+            params.clone(),
+            params.local.our_id,
+            broadcast_public_keys,
+            broadcast_sk,
+            module_cfgs,
+            code_version_str,
+        );
+
+        info!(
+            target: LOG_NET_PEER_DKG,
+            "Comparing consensus config checksum..."
+        );
+
+        let checksum = cfg.consensus.consensus_hash_sha256();
+
+        connections
+            .send(Recipient::Everyone, DkgPeerMessage::Checksum(checksum))
+            .await;
+
```

Comment:
That's some rich statefulness right here :stuck_out_tongue_winking_eye: we are using the same message type for two very different checksums depending where in the protocol we are.